plan/plan.go

```diff
@@ -146,23 +146,54 @@ func (p *Plan) Calculate() *Plan {
 
 	for _, topRow := range t.rows {
 		for _, row := range topRow {
-			if row.current == nil { //dns name not taken
-				changes.Create = append(changes.Create, t.resolver.ResolveCreate(row.candidates))
+			RecordTypeAAAA := []*endpoint.Endpoint{}
+			RecordTypeA := []*endpoint.Endpoint{}
```

Comment:
It would be better to change `planTable.rows` to be a `map[string]map[string]map[string]*planTableRow` and separate the rows by `RecordType`. That would make this code be simpler and clearer, with less duplication. 

It would also handle deleting only one record type.

Reply:
Could you please explain what you mean exactly? I'm not sure how I would use such a map.

Reply:
`addCurrent()` and `addCandidate()` would create a different `planTableRow` for each value of `endpoint.RecordType`, indexed by the `endpoint.RecordType` value.

Reply:
I'm not following as to how I would do that? Would it be possible for you to make the needed changes for addCurrent and AddCandidate? I can then figure out the main Calculate logic for it?

Reply:
I sent you a PR